### DISCUSSION TITLE
How to configure Ghostfolio to use DataSources other than YAHOO & MANUAL?

### DISCUSSION THREAD

**Discussion by sameleff** [ghostfolio/ghostfolio#2606] [Q&A]
I see from the schema.prisma file that there are 8 different available DataSource types that can be configured. What are the steps to enable one of these DataSources within Ghostfolio? For example:
1. I assume we will need to create and pass additional environment variables for things such as API keys to enable certain DataSource types. Are there standard names for these variables? If so, where can I find the list of these additional variables? For example, from #2090 I see a reference to a `DATA_SOURCES` environment variable...but this is not documented in the [.env.example](https://github.com/ghostfolio/ghostfolio/blob/main/.env.example) file. Is there a list of all environment variables that I can reference?
2. My understanding is thus: Ghostfolio will automatically query all DataSource types and add pricing data to Ghostfolio (except for type MANUAL and YAHOO). Please correct me if I am misunderstanding.
3. Is there a way to assign a DataSource type when importing activity (manually, using the GUI importer, or the API)?
4. Can you assign / change a DataSource assignment in the Admin Control panel Market Data tab? If so, how?

**Comment by dtslvr** [comment]
Hi @sameleff

Of course you can specify the data source in the data you import (see [example](https://github.com/ghostfolio/ghostfolio/blob/main/test/import/ok.csv)).

The officially supported data sources are preconfigured by default (`COINGECKO`, `MANUAL` and `YAHOO` at this point). However, all other data sources are considered experimental and subject to change or be removed, so you may use them at your own discretion and risk.

**Reply by sameleff** [reply]
Where do I find references for the environment variables I need to configure in order to enable the other data sources (other than the `DATA_SOURCES = ['MANUAL', 'YAHOO', 'COINGECKO']` one)?

**Reply by coharms** [reply]
You have to dig in the code a bit. In the data-provider directory ([here](https://github.com/ghostfolio/ghostfolio/tree/d155ab6f28935b2f4b0f36f8f49339ad382f752b/apps/api/src/services/data-provider)) you can find the implementations for each of the available data sources. Each service.ts will probably contain the querying of the environment variable from the `ConfigurationService`. The said environment variable then has to be filled in your `.env` file. 
For the Alpha Vantage data source ([implemented here](https://github.com/ghostfolio/ghostfolio/blob/d155ab6f28935b2f4b0f36f8f49339ad382f752b/apps/api/src/services/data-provider/alpha-vantage/alpha-vantage.service.ts)) the environment variable is `ALPHA_VANTAGE_API_KEY`. That way I managed to get Alpha Vantage to at least list Assets, however beyond that the data source did not work for me.

**Reply by sameleff** [reply]
Aha! I see that all now, thanks very much for the reply! Curiously, for the CoinGeckoService, I don't see a `ConfigurationService`...do we need to create an account and obtain our own coingecko API key, or does the ghostfolio code already connect to coingecko using a generic api key?

**Reply by coharms** [reply]
As mentioned coingecko is an officially supported data source, therefore no further configuration is required, fetching crypto assets just works out of the box. I can also not find any key initially loaded or sent with the requests, therefore I suppose that no key is needed.

**Reply by sameleff** [reply]
Wonderful, that's what I thought but good to have it confirmed. Thanks for the help here!
